src/query/api/v1/handler/fetch_options.go

```diff
@@ -66,20 +66,42 @@ func NewFetchOptionsBuilder(
 	return fetchOptionsBuilder{opts: opts}
 }
 
-func (b fetchOptionsBuilder) NewFetchOptions(
-	req *http.Request,
-) (*storage.FetchOptions, *xhttp.ParseError) {
-	fetchOpts := storage.NewFetchOptions()
-	fetchOpts.Limit = b.opts.Limit
+// ParseLimit parses request limit from either header or query string.
+func ParseLimit(req *http.Request, defaultLimit int) (int, error) {
 	if str := req.Header.Get(LimitMaxSeriesHeader); str != "" {
 		n, err := strconv.Atoi(str)
 		if err != nil {
 			err = fmt.Errorf(
 				"could not parse limit: input=%s, err=%v", str, err)
-			return nil, xhttp.NewParseError(err, http.StatusBadRequest)
+			return 0, err
+		}
+
+		return n, nil
+	}
+
+	if str := req.URL.Query().Get("limit"); str != "" {
+		n, err := strconv.Atoi(str)
+		if err != nil {
+			err = fmt.Errorf(
+				"could not parse limit: input=%s, err=%v", str, err)
+			return 0, err
 		}
-		fetchOpts.Limit = n
+
+		return n, nil
+	}
+
+	return defaultLimit, nil
+}
+
+func (b fetchOptionsBuilder) NewFetchOptions(
```

Comment:
Does `NewFetchOptions` need a comment? Weird that the lint isn't catching that.

Reply:
Sure sounds good; I think linter is currently disabled for whatever reason (probably all the linting errors :p)